src/engine/routing_algorithms/many_to_many.cpp

```diff
@@ -132,11 +132,16 @@ inline LevelID getNodeQueryLevel(const MultiLevelPartition &partition,
         return INVALID_LEVEL_ID;
     };
 
-    return std::min(highest_diffrent_level(phantom_node.forward_segment_id),
-                    highest_diffrent_level(phantom_node.reverse_segment_id));
+    const auto node_level = std::min(highest_diffrent_level(phantom_node.forward_segment_id),
```

Comment:
Shouldn't a phantom node be in the same cell with all of it's segments? Why do we need both comparisons here?

Reply:
it is not guaranteed in EBG after renumbering and re-partitioning of EBG nodes. I can place an assertion to check the condition, but would like to keep atm check for forward and reverse segments.

Reply:
In case we actually do not put the same node into the same partition, I would say the check is the better way to go. 
Is there a reason for not hitting that guarantee, though? Seems to me that the same location should always be in the same cell.

Reply:
i would rather be careful here, EBG nodes of the same NBG edge can be potentially in different cells.
I would like to keep the condition here as `min(forward level, reverse level)` as in the rest of the code and check the assumption in a separate issue/pr.

Reply:
@oxidase I am not asking you to change the check here, but asking if you know of a reason we do not keep a single nbg edge in a single cell. Separating a single edge logically into two different cells seems like a strange concept to me and something we might need to rethink.

Reply:
i don't quite understand the question "... keep a single nbg edge in a single cell" . The reason for doing that is in using  an NBG partition to make an EBG one with some heuristics  how to place EBG boundary nodes in cells and this potentially could lead to the situation  "Separating a single edge logically into two different cells". This is not really optimal as also can result in "invalid" boundary nodes that neither sources nor destinations, but number of such nodes is vanishingly small. 

I would suggest to open an issue with ideas how to reimplement this, because i still don't understand benefits of doing NBG  instead of EBG partitioning.

Reply:
Opening an issue for this seems fine with me, the only actual issue that needs to change is the introduction of the template parameter below, since it is unrelated to CH.